StorageUtility/Linux/StorageUtility.cpp

```diff
@@ -15,253 +15,185 @@ See the License for the specific language governing permissions and
 limitations under the License.
 </License>
 */
-
+#include <iostream>
 #include <libudev.h>
+#include <fcntl.h>
+#include <map>
 #include <memory>
 #include <string.h>
+#include <unistd.h>
 
-#include "StorageUtility.h"
 #include "Device.h"
+#include "StorageUtility.h"
 
 namespace vtStor
 {
-
-eErrorCode GetStorageDevices(std::vector<std::shared_ptr<cDeviceInterface>>& Devices, eOnErrorBehavior OnErrorBehavior)
+struct s_MappingSystemPathAndDeviceHandle
 {
-    eErrorCode error = eErrorCode::None;
+    std::map<String, vtStor::sStorageAdapterProperty> s_MappingSystemPathToProperties;
+    std::map<DeviceHandle, vtStor::String>  s_MappingDeviceHandleToSysPath;
+};
 
-    sEnumerateDevicesCallback callBack;
-    callBack.Data = (void*)&Devices;
+std::unique_ptr<s_MappingSystemPathAndDeviceHandle>  MappingSystemPathAndDeviceHandle;
 
-//    callBack.Function = [](void* Data, const HDEVINFO& DevInfoHandle, SP_DEVINFO_DATA& DevInfoData, SP_DEVICE_INTERFACE_DATA& DevInterfaceData, const PSP_INTERFACE_DEVICE_DETAIL_DATA& DevDetailData, U32 SizeOfDevDetailData, eErrorCode& ErrorCode)
-//    {
-//        std::vector<std::shared_ptr<vtStor::cDeviceInterface>>* devices = (std::vector<std::shared_ptr<vtStor::cDeviceInterface>>*)Data;
-//        devices->push_back(std::make_shared<cDevice>((SP_DEVINFO_DATA*)&DevInfoData, (SP_DEVICE_INTERFACE_DATA*)&DevInterfaceData, DevDetailData, SizeOfDevDetailData));
-//    };
-
-//    error = EnumerateDevices(callBack, &GUID_DEVINTERFACE_DISK, OnErrorBehavior);
-    return(error);
+eErrorCode GetStorageDevices(std::vector<std::shared_ptr<cDeviceInterface>>& Devices, eOnErrorBehavior OnErrorBehavior)
+{
+    MappingSystemPathAndDeviceHandle = std::make_unique<s_MappingSystemPathAndDeviceHandle>();
+    std::vector<String> DevicePaths;
+    vtStor::GetStorageDevicePaths(DevicePaths, eOnErrorBehavior::Continue);
+    for (int i = 0; i < DevicePaths.size(); ++i)
+    {
+        std::shared_ptr<vtStor::cDevice> Device = std::make_shared<cDevice>(DevicePaths[i]);
+        Devices.push_back(Device);
+    }
+    return(eErrorCode::None);
```

Comment:
I will change return to void, and if we need return a error code for GetStorageDevices() then I will edit, because if GetStorageDevicePaths() return a error code then DevicePaths.size() is 0.